compiler/src/iree/compiler/DispatchCreation/PropagateEncodings.cpp

```diff
@@ -46,57 +46,58 @@ struct PropagateEncodingsPass
 
 LogicalResult SwapEncodingOpWithTensorCollapseShapeOp::matchAndRewrite(
     IREE::Encoding::SetEncodingOp encodingOp, PatternRewriter &rewriter) const {
+
+  Value target = encodingOp.getSource();
+
   auto encoding = dyn_cast<IREE::Encoding::MatmulKAttr>(
       encodingOp.getResultType().getEncoding());
-  if (!encoding) {
-    return rewriter.notifyMatchFailure(encodingOp, "only matmul_k is handled");
+
+  auto attrPropInterface =
+      llvm::dyn_cast_or_null<IREE::Encoding::EncodingPropagationAttrInterface>(
+          encoding);
+
+  if (!attrPropInterface || !attrPropInterface.isPropagable(target)) {
+    return failure();
+  }
```

Comment:
By op semantic, the result type of the SetEncoding op should always have an encoding. Thus, you can use `dyn_cast`.

Style nits:
1. Do not add a blank line in the start of functions, and please delete other blank lines in this function. It makes more sense to me to not have a blank line right before the checks. They are highly relevant. https://google.github.io/styleguide/cppguide.html#Vertical_Whitespace
2. you can delete `encoding` at all I think?